File: Week 1 C - Basics/cash/cash.c
#include <cs50.h>
#include <stdio.h>

#define QUARTER 25
#define DIME 10
#define NICKEL 5
#define PENNY 1

int change_owed(void);
int amount_of_coins(int changeOwed);

int main(void)
{

    int changeOwed = change_owed();
    int coins = amount_of_coins(changeOwed);
    printf("%i\n", coins);
}

int amount_of_coins(int changeOwed)
{
    int amountOfCoins = 0;

    if (changeOwed >= QUARTER)
    {
        amountOfCoins += changeOwed / QUARTER;
        changeOwed = changeOwed % QUARTER;
    }

    if (changeOwed >= DIME)
    {
        amountOfCoins += changeOwed / DIME;
        changeOwed = changeOwed % DIME;
    }

    if (changeOwed >= NICKEL)
    {
        amountOfCoins += changeOwed / NICKEL;
        changeOwed = changeOwed % NICKEL;
    }

    if (changeOwed >= PENNY)
    {
        amountOfCoins += changeOwed / PENNY;
        changeOwed = changeOwed % PENNY;
    }

    return amountOfCoins;
}

int change_owed(void)
{

    int changeOwed;

    do
    {
        changeOwed = get_int("change_owed: \n");
        if (changeOwed <= 0)
        {
            printf("You can't owe 0 or less. \n");
        }
    }
    while (changeOwed <= 0);
    return changeOwed;
}

File: Week 6 Python/sentimental-cash/cash.py
from cs50 import get_float

QUARTER = 25
DIME = 10
NICKEL = 5
PENNY = 1


def main():

    changeOwed = round(change_owed() * 100)
    coins = amount_of_coints(changeOwed)
    print(coins)


def amount_of_coints(changeOwed):

    amountOfCoins = 0

    # in every iterarion of this for "coin" takes the value of each constant (quarter = 25...)
    # if we were using "for coin in range(4)", then coin would take the values 0, 1, 2, 3
    for coin in [QUARTER, DIME, NICKEL, PENNY]:
        amountOfCoins += changeOwed // coin
        changeOwed %= coin

    return amountOfCoins


def change_owed():
    while True:
        changeOwed = get_float("Change owed: ")
        if changeOwed <= 0:
            print("You can't owe 0 or less")
        else:
            return changeOwed


if __name__ == "__main__":
    main()
